examples/problems/co2ptflashproblem.hh

```diff
@@ -498,6 +498,25 @@ private:
             fs.setViscosity(FluidSystem::gasPhaseIdx, FluidSystem::viscosity(fs, paramCache, FluidSystem::gasPhaseIdx));
         }
 
+        // determine the component total fractions
+        // TODO: duplicated code here, while should be refactored out when we swithing
+        // to starting from total mole fractions
+        Dune::FieldVector<Scalar, numComponents> z(0.0);
+        Scalar sumMoles = 0.0;
+        for (unsigned phaseIdx = 0; phaseIdx < numPhases; ++phaseIdx) {
+            const auto saturation = getValue(fs.saturation(phaseIdx));
+            for (unsigned compIdx = 0; compIdx < numComponents; ++compIdx) {
+                Scalar tmp = getValue(fs.molarity(phaseIdx, compIdx) * saturation);
```

Comment:
If you write `getValue(fs.molarity(...)) * saturation` instead of `getValue(fs.molarity(...) * saturation)` then the resulting operation is never more expensive than what's here.  The former is `Scalar * Scalar` whereas the latter is `getValue(something * Scalar)` which **could** be more expensive.